draft-irtf-cfrg-rsa-blind-signatures.md

```diff
@@ -338,6 +358,120 @@ Steps:
+
+Applications that provide high-entropy input messages can expose the internal
+rsabssa_blind and rsabssa_finalize directly, as the additional message randomization
+does not offer security advantages. See {{Lys22}}, {{apis}}, and {{message-entropy}}
+for more information.
+
+### Salted Blind
+
+rsabssa_salted_blind invokes rsabssa_blind with a salted input message and outputs the
+blinded message to be sent to the server and the corresponding inverse, both encoded
+as octet strings, as well as the fresh message salt, which is 32 random bytes.
+
+~~~
+rsabssa_salted_blind(pkS, msg)
+
+Parameters:
+- kLen, the length in octets of the RSA modulus n
+- kBits, the length in bits of the RSA modulus n
+- HF, the hash function used to hash the message
+- MGF, the mask generation function
+
+Inputs:
+- pkS, server public key (n, e)
+- msg, message to be signed, an octet string
+
+Outputs:
+- blinded_msg, an octet string of length kLen
+- inv, an octet string of length kLen
+- msg_salt, an octet string of length 32 bytes
+
+Errors:
+- "message too long": Raised when the input message is too long.
+- "encoding error": Raised when the input message fails encoding.
+- "invalid blind": Raised when the inverse of r cannot be found.
+
+Steps:
+1. msg_salt = random(32)
+2. salted_msg = msg_salt || msg
+3. blinded_msg, inv = blind(pkS, salted_msg)
+4. output msg_salt, blinded_msg, inv
+~~~
+
+### Salted Finalize
+
+rsabssa_salted_finalize invokes rsabssa_finalize directly with the salted
+message and outputs the result.
+
+~~~
+rsabssa_salted_finalize(pkS, msg, blind_sig, inv)
+
+Parameters:
+- kLen, the length in octets of the RSA modulus n
+
+Inputs:
+- pkS, server public key (n, e)
+- msg, message to be signed, an octet string
+- msg_salt, the 32 octets random salt used to salt the message
+- blind_sig, signed and blinded element, an octet string of
+  length kLen
+- inv, inverse of the blind, an octet string of length kLen
+
+Outputs:
+- sig, an octet string of length kLen
+
+Errors:
+- "invalid signature": Raised when the signature is invalid
+- "unexpected input size": Raised when a byte string input doesn't
+  have the expected length.
+
+Steps:
+1. salted_msg = msg_salt || msg
+2. output rsabssa_finalize(pkS, salted_msg, blind_sig, inv)
+~~~
+
+### Salted Verify
+
+rsabssa_salted_verify validates the resulting unblinded signature computed over a
+salted message. It invokes RSASSA-PSS-VERIFY directly by augmenting the input
+message with the message salt.
+
+~~~
+rsabssa_salted_verify(pkS, msg, msg_salt, sig)
+
+Parameters:
+- kLen, the length in octets of the RSA modulus n
+
+Inputs:
+- pkS, server public key (n, e)
+- msg, message to be signed, an octet string
+- msg_salt, the 32 octets random salt used to salt the message
+- sig, signature of the salted_msg
+
+Outputs:
+- "valid signature" if the signature is valid
+
+Errors:
+- "invalid signature": Raised when the signature is invalid
+
+Steps:
+1. salted_msg = HF(salt_msg || msg )
```

Comment:
If we decide to not have the extra hashing, then this is wrong. Should not have HF()